app/jobs/data_cleanup_job.rb

```diff
@@ -0,0 +1,18 @@
+class DataCleanupJob < ApplicationJob
+
+  queue_as :default
+
+  def perform
+    threshold = 90.days.ago
+
+    user_ids = Message.
```

Comment:
I like `pluck` a lot for these cases, because unlike select it doesn't construct full ActiveRecord objects with pruned attributes, but rather just creates a simple lightweight array, but still with all the expected typecasting. And then also you don't have to `to_a`...`map`. But that's really performance golfing, and not a blocking thing. Just thought I'd mention it.

Reply:
Great suggestions - it reads so much better now!